### ISSUE TITLE
Provide verbose user information

### ISSUE THREAD

**Issue by Miconen** [Miconen/tectonic-api#17]
### Verbose user information should include everything required in the /profile bot command.
tldr: We need to query the DB for the required data for the `DetailedUser` struct in `database/utils.go`

Current usage is defined at `database/detailed_users.go` in the `GetUsers()` function. The other functions reciding in `database/detailed_users.go` should also adopt this "verbose" response but this shouldn't be that much harder once we figure out the SQL query side.

> [!WARNING]
> Current implementation of the `getUsers` query works to some extent but looks absolutely horrid and doesn't accept parameters.

### TODO and issues i faced:
- [x] Query data from multiple tables
- [x] `DetailedTime` array includes an array of `UserData` which i struggled to figure out
- [x] Casting the data into the corresponding format
- [x] Preferably using a more maintainable and readable query than currently used

**Comment by Miconen** [comment]
To clarify, this is marked as a blocker because we have existing functionality that relies on this. (Profile command)
